src/metadataGeneration/typeResolver.ts

```diff
@@ -298,7 +298,8 @@ export class TypeResolver {
 
     if (extractEnum) {
       const enums = enumDeclaration.members.map((member: any, index) => {
-        return getEnumValue(member) || String(index);
```

Comment:
@fhewitt here's a step by step instruction on how to write a unit test for this:

1) add an enum member to https://github.com/lukeautry/tsoa/blob/13d856e68fd1349c3db98ff1160645f1ae8c27e6/tests/fixtures/testModel.ts#L131

So when you're done, it should look like:
```ts
export enum EnumStringValue {
  EMPTY = '',
  VALUE_1 = 'VALUE_1' as any,
  VALUE_2 = 'VALUE_2' as any,
}
```

2) Do the same for the number enums so we know that this PR doesn't create a regression bug. So add a 0 case for EnumNumberValue here: https://github.com/lukeautry/tsoa/blob/13d856e68fd1349c3db98ff1160645f1ae8c27e6/tests/fixtures/testModel.ts#L123

```ts
export enum EnumNumberValue {
  VALUE_0 = 0,
  VALUE_1 = 2,
  VALUE_2 = 5,
}
```

3) Update the unit tests to check the actual data that is produced from the schema. I would recommend a chai assertion like:
```ts
const expectedEnumValues = ['', 'VALUE_1', 'VALUE_2'];
expect(propertySchema.enum).to.eq(true, `for property ${propertyName}[enum]`);
```

3a) update the string enum tests: 
* for OpenAPI 3: https://github.com/lukeautry/tsoa/blob/e482d3c9325c4c11a376ca096628c388c64dd8ca/tests/unit/swagger/schemaDetails3.spec.ts#L303
* for Swagger 2: https://github.com/lukeautry/tsoa/blob/13d856e68fd1349c3db98ff1160645f1ae8c27e6/tests/unit/swagger/definitionsGeneration/definitions.spec.ts#L248

3b) update the nnumber enum tests:
* for OpenAPI 3: https://github.com/lukeautry/tsoa/blob/13d856e68fd1349c3db98ff1160645f1ae8c27e6/tests/unit/swagger/schemaDetails3.spec.ts#L288
* for Swagger 2: https://github.com/lukeautry/tsoa/blob/13d856e68fd1349c3db98ff1160645f1ae8c27e6/tests/unit/swagger/definitionsGeneration/definitions.spec.ts#L264

Reply:
Ok, so I were close, but it fail:

```expect(propertySchema.enum).to.eq(expectedEnumValues, `for property ${propertyName}[enum]`);```

give

```
Definition generation for OpenAPI 3.0.0
       for specDefault
         should set additionalProperties to false if noImplicitAdditionalProperties is set to "throw-on-extras" (when there are no dictionary or any types)
           should produce a valid schema for the enumStringArray property on TestModel for the specDefault:
     **AssertionError: for property enumStringArray[enum]: expected undefined to equal [ '', 'VALUE_1', 'VALUE_2' ]**
      at Object.enumStringArray (tests\unit\swagger\schemaDetails3.spec.ts:315:44)
      at Context.<anonymous> (tests\unit\swagger\schemaDetails3.spec.ts:498:49)```

Notice that the added test fail also on the master branch.

The object receive:
```
{ items: { '$ref': '#/components/schemas/EnumStringValue' },
  type: 'array',
  description: undefined,
  format: undefined,
  default: undefined,
  nullable: true
}
```

Template helper use the `property.enums` value for the function validateEnum. Not sure why this isn't generated for the test...

Reply:
Enums create a ref type.
@fhewitt you can use `getValidatedDefinition('EnumStringValue', currentSpec);` to get the definition and proceed to do the checks @dgreene1 mentioned on the definition

Reply:
Mmmm, this method didn't exist in schemaDetails3.specs.ts.

I go with the hypothesis that the validation check didn't depends on the schema version, and therefore that schemaDetails3.spec.ts doesn't require any change. If I'm right, the latest commit should cover the unit test requirement.

Thank you both for you explanation and instructions.

Reply:
You're still missing the tests for OpenAPI 3. It's unfortunate that there isn't a `getValidatedDefinition` in the OpenAPI 3 test file. But you can replicate it by checking the structure of `specDefault.spec.components.schemas.EnumStringValue`

Thank you for the additional tests and for submitting this PR. I'd really like to see coverage for OpenAPI 3 before we merge this PR since most people will probably be using OpenAPI 3 (since Swagger 2 doesn't support the majority of features).

Reply:
At the same time, a *lot* of unit test over the validation are present in definitions.spec.ts but missing in OpenAPI 3. I strongly suspect that it's because the unit tests over validation are good for both cases / all OpenAPI 3 versions, and therefore not required to be duplicated. I can do it (but not today, I fear) if really required.

Reply:
I don’t believe that the missing tests in the OpenAPI test file are intentional. I believe they are missing because OpenAPI 3 was only added to tsoa about 3 months ago. So the test coverage was minimal because it was new. You can even see that in the PR for that file where the original author was like “I hope this works?”

So yes it would be really appreciated if you add the tests. Thank you for your patients. I know that it seems unnecessary but it will really help us maintainers and contributes. :)

Reply:
Sprint delivered, back here!

You seems right, I've remade an equivalent of `getValidatedDefinition` and it required enough changes that it's seems definitely to require distinct testing.

Let me know if you see something else.